core/pom.xml

```diff
@@ -30,6 +30,10 @@
          <artifactId>jgroups</artifactId>
       </dependency>
 
+      <dependency>
```

Comment:
hmmm maybe we need something better...

How about creating a separate module (`infinispan-cloud`), placing all "cloud related" configurations there and adding all discovery protocol dependencies (not only Kubernetes but probably Azure as well (in the future)).

This way users would add only one dependency to their classpath (`infinispan-cloud`) and have all necessary dependencies and configurations in place...